packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -150,20 +162,50 @@ export class AccountTrackerController extends BaseController<
   }
 
   /**
-   * Refreshes all accounts in the current keychain.
+   * Refreshes the balances of the accounts depending on the multi-account setting.
+   * If multi-account is disabled, only updates the selected account balance.
+   * If multi-account is enabled, updates balances for all accounts.
+   *
+   * @async
    */
   refresh = async () => {
     this.syncAccounts();
     const accounts = { ...this.state.accounts };
+    const isMultiAccountBalancesEnabled = this.getMultiAccountBalancesEnabled();
+
+    if (!isMultiAccountBalancesEnabled) {
+      const selectedAddress = this.getSelectedAddress();
+      const balance = await this.getBalanceFromChain(selectedAddress);
+      accounts[selectedAddress] = { balance: BNToHex(balance) };
+      this.update({ accounts });
+      return;
+    }
+
     for (const address in accounts) {
```

Comment:
Nit: Maybe it would be simpler to create an "accounts to update" list, and reuse the same logic for both of these cases?

```suggestion
    const accountsToUpdate = isMultiAccountBalancesEnabled
      ? Object.keys(accounts)
      : [selectedAddress];

    for (const address of accountsToUpdate) {
```

Just a suggestion